### PR TITLE
chore(core,ext): Minor JS optimisations

### PR DESCRIPTION
**Pull request by aapoalas**
Some exceedingly minor JS optimisations:
1. Avoid creating N objects when initialising `opIds` where N is the number of ops. 1 object will do.
2. Avoid a few other unnecessary object spreads, one in `Datagram.prototype.send` and another in a webidl `makeException` call. There are quite a bit more of object spreads around here that I would like to take a stab at, replacing object spreads with explicit option picking but that shall wait for a better day.
3. Remove `opts` parameter usage from webidl `toNumber` helper function as unused.

### CODE REVIEW

File: ext/webidl/00_webidl.js [hunk 1]
```diff
@@ -328,7 +328,7 @@
   };
 
   converters["unrestricted float"] = (V, opts) => {
```
Comment (kt3k) [inline ext/webidl/00_webidl.js:330]: This `opts` seems unused now

File: ext/webidl/00_webidl.js [hunk 2]
```diff
@@ -714,7 +714,7 @@
           throw makeException(
             TypeError,
             `can not be converted to '${name}' because '${key}' is required in '${name}'.`,
-            { ...opts },
+            { context: opts.context, prefix: opts.prefix },
```
Comment (kt3k) [inline ext/webidl/00_webidl.js:717]: Does it work to pass simply `opts` instead of creating a new object?

File: ext/webidl/00_webidl.js (hunk 1 above)
Comment (aapoalas) [reply ext/webidl/00_webidl.js:330]: Ah, missed this one. `unrestricted double` already got `opts` turned to `_opts` but this one remains.

File: ext/webidl/00_webidl.js (hunk 2 above)
Comment (aapoalas) [reply ext/webidl/00_webidl.js:717]: It definitely should and there are examples of doing exactly that a few lines earlier, but I'm afraid of making functional changes to the code.

That being said, this whether or not this copying is done cannot be a functional change I don't think. I'll fix this into a simple pass.

**Comment by bartlomieju** [comment]
Please rebase with `main` branch

**Review by kt3k — APPROVED** [review]
LGTM

**Comment by aapoalas** [comment]
Merged with main once again.

**Pull request merged**